inference-engine/tests/functional/plugin/cpu/bfloat16/memory_conv.cpp

```diff
@@ -0,0 +1,203 @@
+            <input>
+                <port id="0">
+                    <dim>1</dim>
+                    <dim>200</dim>
+                </port>
+            </input>
+        </layer>
+       <layer id="6" name="Fc_6" precision="FP32" type="InnerProduct"  >
+            <data out-size="200" />
+            <input>
+                <port id="0">
+                    <dim>1</dim>
+                    <dim>200</dim>
+                </port>
+            </input>
+            <output>
+                <port id="1">
+                    <dim>1</dim>
+                    <dim>200</dim>
+                </port>
+            </output>
+            <weights offset="160800" size="160000" />
+            <biases offset="320800" size="800" />
+        </layer>
+    </layers>
+    <edges>
+        <edge from-layer="0" from-port="0" to-layer="2" to-port="0" />
+        <edge from-layer="1" from-port="0" to-layer="2" to-port="1" />
+        <edge from-layer="2" from-port="2" to-layer="3" to-port="0" />
+        <edge from-layer="3" from-port="1" to-layer="4" to-port="0" />
+        <edge from-layer="4" from-port="1" to-layer="5" to-port="0" />
+        <edge from-layer="4" from-port="1" to-layer="6" to-port="0" />
+    </edges>
+</net>
+)V0G0N";
+    InferenceEngine::Blob::Ptr net_bin_v6;
+};
+
+TEST_P(MemoryConv, inferV6Model) {
+    if (!InferenceEngine::with_cpu_x86_bfloat16()) {
+        // on platforms which do not support bfloat16, we are disabling bf16 tests since there are no bf16 primitives,
+        // tests are useless on such platforms
+        return;
+    }
+
+    auto ie = InferenceEngine::Core();
+    auto net = ie.ReadNetwork(net_xml_v6, net_bin_v6);
+    auto exe_net = ie.LoadNetwork(net, "CPU");
+    auto inf_reg = exe_net.CreateInferRequest();
+
+    CNNNetwork info = exe_net.GetExecGraphInfo();
+
+    auto cc = inf_reg.GetPerformanceCounts();
+    std::string mem0_str = "Memory_1";
+    std::string mem1_str = "Memory_5";
+    std::string exp_prec_str = "BF16";
+
+    auto mem0 = std::find_if(cc.begin(), cc.end(), [mem0_str] (const std::pair<std::string, InferenceEngineProfileInfo> & c) -> bool {
+        return c.first.substr(0, mem0_str.length()) == mem0_str;;
+    });
+    auto mem1 = std::find_if(cc.begin(), cc.end(), [mem1_str] (const std::pair<std::string, InferenceEngineProfileInfo> & c) -> bool {
+        return c.first.substr(0, mem1_str.length()) == mem1_str;;
+    });
+    std::string execType0 = mem0->second.exec_type;
+    std::string pfPrecision0 = execType0.substr(execType0.length() - exp_prec_str.length(), exp_prec_str.length());
+    std::string execType1 = mem1->second.exec_type;
+    std::string pfPrecision1 = execType1.substr(execType1.length() - exp_prec_str.length(), exp_prec_str.length());
+    ASSERT_EQ(pfPrecision0, exp_prec_str);
+    ASSERT_EQ(pfPrecision1, exp_prec_str);
+}
+}  // namespace LayerTestsDefinitions
+
+namespace {
+using namespace LayerTestsDefinitions;
```

Comment:
Why `LayerTestDefinitions`?

Reply:
Because test case name is used without namespace specification a several lines below. 
Anyway that's not elegant, so I will change it.

Reply:
This test was rewritten with using Ngraph.